router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go

```diff
@@ -0,0 +1,249 @@
+package marketobulkupload
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+)
+
+type MarketoAPIServiceInterface interface {
+	ImportLeads(csvFilePath, deduplicationField string) (string, *APIError)
+	PollImportStatus(importId string) (*MarketoResponse, *APIError)
+	GetLeadStatus(url string) ([]map[string]string, *APIError)
+}
+
+type MarketoAPIService struct {
+	logger       logger.Logger
+	statsFactory stats.Stats
+	httpClient   *http.Client
+	munchkinId   string
+	authService  MarketoAuthServiceInterface
+	maxRetries   int
+}
+
+type APIError struct {
+	StatusCode int64
```

Comment:
We can probably change it to int and avoid typeCasting al over?

Reply:
Updated